src/test/java/spoon/test/support/ResourceTest.java

```diff
@@ -90,4 +90,31 @@ public void testFilteringFolder() throws Exception {
 		assertEquals(nbAll - nbPackageProtected, mb3.getFactory().getModel().getAllTypes().size());
 	}
 
+	@Test
+	public void testPerformanceOfVirtualFolder() throws Exception {
+		String dir = "src";
+		FileSystemFolder fileSystemFolder = new FileSystemFolder(new File(dir));
+
+		SpoonFolder folder = new VirtualFolder();
+		folder.addFolder(fileSystemFolder);
+
+		//Measure how long it needs to return all files
+		long startTime = System.currentTimeMillis();
+		folder.getAllFiles();
+		assertTrue("The VirtualFolder.getAllFiles() is too slow", (System.currentTimeMillis() - startTime)<1000);
```

Comment:
In Spoon, we try not to make assertions based on currentTimeMillis, they are too dependent on the platform. To me, you can remove the tests. Everybody agrees?